src/generators/web.cr

```diff
@@ -10,8 +10,13 @@ class LuckyCli::Generators::Web
     project_name : String,
     @options : Options
   )
-    @project_dir = project_name.gsub(" ", "_")
-    @project_name = @project_dir.gsub("-", "_")
+    @project_dir = project_name
+    @project_name = project_name
+
+    validate(directory_name: @project_dir)
+    validate(project_name: @project_name)
```

Comment:
This is now not doing any manipulation of user input. If it's good it passes and if it's not it fails. I have a hunch that may not be the most helpful, but I'm not sure where to draw that line.